src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -262,47 +234,19 @@ private async Task ProcessQueueAsync()
 
                 // If there was an in progress work item that failed in the queue logic, set the result of the queue item
                 // to the exception so that it bubbles back to the caller.
-                inProgressWorkItem?.HandleQueueFailure(e);
+                inProgressWorkItem?.TrySetException(e);
```

Comment:
i'd like to talk about this.  i think it's necessary, but i want to understand it better.